tls/extensions/s2n_client_supported_versions.c

```diff
@@ -121,7 +121,7 @@ static int s2n_extensions_client_supported_versions_process(struct s2n_connectio
 
 static int s2n_client_supported_versions_recv(struct s2n_connection *conn, struct s2n_stuffer *in)
 {
-    if (!s2n_is_tls13_enabled()) {
+    if (s2n_connection_get_protocol_version(conn) < S2N_TLS13) {
```

Comment:
what does s2n_connection_get_protocol_version() return? if it isn't tls1.3, this function would not run and version wouldn't be upgraded?

Reply:
get_protocol_version() returns actual_protocol_version if set, or client/server_protocol_version if not: https://github.com/awslabs/s2n/blob/bd5aaa12874d1a18d3bfba174c708041423d950a/tls/s2n_connection.c#L1311-L1325

We shouldn't process TLS1.3 extensions on a non-TLS1.3 connection. Calling s2n_enable_tls13() sets the connection client/server_protocol_version to TLS1.3.